roles/confluent.ksql/tasks/rbac.yml

```
@@ -1,4 +1,8 @@
 ---
+- name: Get Principal used on the Inter Broker Listener
+  include_tasks: set_principal.yml
+  when: ksql_log_streaming_enabled | bool
+
```

The set_principal.yml file looks like duplicating code, itd be cleaner to do this:

```suggestion
- name: Get Principal used on the Inter Broker Listener
  include_role: 
    name: confluent.kafka_broker
    tasks_from: set_principal.yml
  when: ksql_log_streaming_enabled | bool

- set_fact:
    ksql_log4j_principal: kafka_broker_principal
  when: ksql_log_streaming_enabled | bool
```

---

or there isnt a need for the ksql_log4j_principal variable, you could just use the var kafka_broker_principal down below

---

I just realized to make this work in the referenced set_principal.yml it would need the ability to pass a keystore path variable, so like this:
```
- name: Extract Common Name from Keystore - SSL Mutual Auth
  # Examine the keystore
  # Search lines with Entry type: "PrivateKeyEntry" and return that line and all after, ca cert is of type "trustedCertEntry"
  # Search for first "Owner" line
  # Extract DNAME from line
  # Remove spaces after commas
  shell: |
    keytool -list -keystore {{keystore_path | default(kafka_broker_keystore_path)}} \
```

and then in your include_role, it would be:
```
- name: Get Principal used on the Inter Broker Listener
  include_role: 
    name: confluent.kafka_broker
    tasks_from: set_principal.yml
  vars:
    keystore_path: ksql_keystore_path
  when: ksql_log_streaming_enabled | bool
```

---

@domenicbove Happy to discuss this further on a call, but initially, I disagree.  I actually did think about this, however came to the conclusion that this should be self contained to the KSQL role, as it is such a specific piece of functionality to KSQL setup due to the unique configuration of this feature.  I think sharing this code could lead to serious problems if it's edited in the future and the context of where it's used is not fully understood.

It generally falls in line with Ansible best practices as well, whereby its recommended roles are self contained.  I say this with the understanding that we already don't follow this best practice in a lot of places. :). I think however, due to how quirky log streaming is, we should keep it separate.  Happy to discuss further though. :)